codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AssembleMiddlewareStack.java

```diff
@@ -0,0 +1,145 @@
+        ).build();
+
+        return ListUtils.of(
+                // Add RequestInvocationIDMiddleware to operation stack
+                RuntimeClientPlugin.builder()
+                    .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                        writer.write("$L.Initialize.Add($T{}, $T)", stackOperand,
+                                RequestInvocationIDMiddleware, middlewareStepAfter);
+                    }).build(),
+
+                // Add serializer middleware to operation stack
+                RuntimeClientPlugin.builder()
+                        .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                            if (protocolGenerator == null){
+                                return;
+                            }
+                            String serializerMiddlewareName = ProtocolGenerator.getSerializeMiddlewareName(operation.getId(),
+                                    protocolGenerator.getProtocolName());
+                            writer.write("$L.Serialize.Add(&$L{}, $T)",
+                                    stackOperand, serializerMiddlewareName, middlewareStepAfter);
+                        })
+                        .build(),
+
+                // Add deserializer middleware to operation stack
+                RuntimeClientPlugin.builder()
+                        .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                            if (protocolGenerator == null) {
+                                return;
+                            }
+                            String deserializerMiddlewareName = ProtocolGenerator.getDeserializeMiddlewareName(operation.getId(),
+                                    protocolGenerator.getProtocolName());
+                            writer.write("$L.Deserialize.Add(&$L{}, $T)",
+                                    stackOperand, deserializerMiddlewareName, middlewareStepAfter);
+                        })
+                        .build(),
+
+                // Add attemptClockSkew middleware to operation stack
+                RuntimeClientPlugin.builder()
+                        .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                            writer.write("$L.Deserialize.Add($T{}, $T)", stackOperand,
+                                    attemptClockSkewMiddleware, middlewareStepAfter);
+                        })
+                        .build(),
+
+                // Add newAttempt middleware to operation stack
+                RuntimeClientPlugin.builder()
+                        .buildMiddlewareStack((writer, service, operation, protocolGenerator, stackOperand) -> {
+                            writer.write("$L.Finalize.Add($T(options.Retryer), $T)", stackOperand,
+                                    newAttemptMiddleware, middlewareStepAfter);
+                        })
```

Comment:
the `NewAttemptMiddleware` shoudl take the request cloner as the second parameter, Doesn't look like the smithy-go/transport/http pkg has this implemented, but it probably should and be used here.